.github/workflows/release_staging.yml

```diff
@@ -0,0 +1,76 @@
+name: Deriv App
+on:
+  push:
+    branches:
+    - master
+env:
+  CACHE_VERSION: ${{ secrets.CACHE_VERSION }}
+  COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_REPO_TOKEN }}
+  DATADOG_CLIENT_TOKEN_LOGS: ${{ secrets.DATADOG_CLIENT_TOKEN_LOGS }}
+  DATADOG_SESSION_REPLAY_SAMPLE_RATE: ${{ secrets.DATADOG_SESSION_REPLAY_SAMPLE_RATE }}
+  DATADOG_SESSION_SAMPLE_RATE: ${{ secrets.DATADOG_SESSION_SAMPLE_RATE }}
+  DATADOG_SESSION_SAMPLE_RATE_LOGS: ${{ secrets.DATADOG_SESSION_SAMPLE_RATE_LOGS }}
+  GD_API_KEY: ${{ secrets.GD_API_KEY }}
+  GD_APP_ID: ${{ secrets.GD_APP_ID }}
+  GD_CLIENT_ID: ${{ secrets.GD_CLIENT_ID }}
+  RUDDERSTACK_PRODUCTION_KEY: ${{ secrets.RUDDERSTACK_PRODUCTION_KEY }}
+  RUDDERSTACK_STAGING_KEY: ${{ secrets.RUDDERSTACK_STAGING_KEY }}
+  RUDDERSTACK_URL: ${{ secrets.RUDDERSTACK_URL }}
+jobs:
+  release_staging:
+    name: Release Staging
+    runs-on: Runner_16cores
+    container:
+      image: node:18.16.0
+    env:
+      NODE_ENV: staging
+      CA: ${{ secrets.CA }}
+      CA_CRT: ${{ secrets.CA_CRT }}
+      DATADOG_APPLICATION_ID: ${{ secrets.DATADOG_APPLICATION_ID }}
+      DATADOG_CLIENT_TOKEN: ${{ secrets.DATADOG_CLIENT_TOKEN }}
+      SERVICEACCOUNT_TOKEN: ${{ secrets.SERVICEACCOUNT_TOKEN }}
+      SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
+    steps:
+    - uses: actions/checkout@v3.6.0
+    - uses: "./.github/actions/git_checkout_from_cache"
+    - uses: "./.github/actions/npm_install_from_cache"
+    - uses: "./.github/actions/build"
+    - uses: "./.github/actions/versioning"
+      with:
+        version_name: staging
+    - uses: actions/upload-artifact@v3.1.1
+      with:
+        path: packages/core/dist
+
+  publish_cloudflare_staging:
+    runs-on: Runner_16cores
+    container:
+      image: node:18.16.0
+    needs:
+    - release_staging
+    env:
+      NODE_ENV: staging
+      CA: ${{ secrets.CA }}
+      CA_CRT: ${{ secrets.CA_CRT }}
+      CLOUDFLARE_ACCOUNT_ID: ${{ secrets.CLOUDFLARE_ACCOUNT_ID }}
+      CLOUDFLARE_API_TOKEN: ${{ secrets.CLOUDFLARE_API_TOKEN }}
+      SERVICEACCOUNT_TOKEN: ${{ secrets.SERVICEACCOUNT_TOKEN }}
+    steps:
+    - uses: actions/checkout@v3.6.0
+    - uses: actions/download-artifact@v3.0.1
+      with:
+        path: packages
+    - uses: "./.github/actions/publish_to_pages_staging"
+
+  send_slack_notification:
+      name: Send Slack notification
+      runs-on: Runner_16cores
+      needs: [build_test_and_upload_artifact, publish_cloudflare_staging]
+      steps:
+      - uses: technote-space/workflow-conclusion-action@v3
```

Comment:
this action gets the overall workflow run conclusion, if any failure happens it will pass the status as failure to the slack action.